agent/cmd/init.go

```diff
@@ -76,61 +71,38 @@ For Flux sources:
 				os.Exit(2)
 			}
 			log.Infow("flux artifact fetched", "dir", TargetDir)
+			return
+		}
+
+		if GitURL == "" {
+			log.Errorw("need --flux-url or --git-url")
+			os.Exit(1)
+
 		}
 
 		// fetch the configured git artifact
-		if GitUrl != "" {
-			u, err := url.Parse(GitUrl)
-			if err != nil {
-				log.Errorw("fatal: unable to parse git url", zap.Error(err))
-				os.Exit(2)
-			}
-			// Configure authentication strategy to access the source
-			authData := map[string][]byte{}
-			authOpts, err := git.NewAuthOptions(*u, authData)
-			if err != nil {
-				log.Errorw("fatal: unable to parse git auth options", zap.Error(err))
-				os.Exit(2)
-			}
-			cloneOpts := repository.CloneConfig{
-				RecurseSubmodules: false,
-				ShallowClone:      true,
-			}
-			cloneOpts.Commit = GitRevision
-			log.Infow("git source fetching", "url", GitUrl, "revision", GitRevision)
-			_, err = gitCheckout(ctx, GitUrl, cloneOpts, authOpts, nil, TargetDir)
-			if err != nil {
-				log.Errorw("fatal: unable to fetch git source", zap.Error(err))
-				os.Exit(2)
-			}
-			log.Infow("git artifact fetched", "dir", TargetDir)
+		auth := &auto.GitAuth{
+			SSHPrivateKey:       os.Getenv("GIT_SSH_PRIVATE_KEY"),
+			Username:            os.Getenv("GIT_USERNAME"),
+			Password:            os.Getenv("GIT_PASSWORD"),
+			PersonalAccessToken: os.Getenv("GIT_TOKEN"),
 		}
-	},
-}
+		repo := auto.GitRepo{
+			URL: GitURL,
+			//ProjectPath: ,
+			CommitHash: GitRevision,
+			Auth:       auth,
+			Shallow:    os.Getenv("GIT_SHALLOW") == "true",
+		}
+
+		_, err = auto.NewLocalWorkspace(ctx, auto.Repo(repo))
```

Comment:
Expecting the code to be checked out the the TargetDir, which is `/share/source`.